transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpProtocolCodecFilter.java

```diff
@@ -58,8 +60,23 @@ public void filterWrite(NextFilter nextFilter,
                 IoBufferEx cachedProtocolBuffer = message.getCache().get(nextProtocol);
                 if (cachedProtocolBuffer != null) {
                     if(cachedProtocolBuffer.capacity()==0) {
-                        writeRequest.getFuture().setWritten();
-                        nextFilter.messageSent(session, writeRequest);
+                        // We must commit this http session if the child is committing, to ensure
+                        // the http headers get written
+                        HttpSession child = HttpAcceptor.SESSION_KEY.get(session);
```

Comment:
`HttpAcceptor.SESSION_KEY` should be parametrically typed in `HttpAcceptSession` where `commit()` is an exposed method, eliminating the need to downcast to `DefaultHttpSession`, right?